java/core/src/java/org/apache/orc/impl/TreeReaderFactory.java

```diff
@@ -1994,7 +1995,18 @@ private static byte[] commonReadByteArrays(InStream stream, IntegerReader length
           totalLength = (int) (batchSize * scratchlcv.vector[0]);
         }
       }
-
+      if (totalLength < 0) {
+        StringBuilder sb = new StringBuilder("totalLength:" + totalLength
+                + " is a negative number.");
+        if (batchSize > 1) {
+          sb.append(" The current batch size is ")
+                  .append(batchSize)
+                  .append(", you can reduce the value by '")
+                  .append(OrcConf.ROW_BATCH_SIZE.getAttribute())
+                  .append("'.");
```

Comment:
Instead of this chaining, the following will be better in the readability.
```java
sb.append(" The current batch size is ")
sb.append(batchSize)
sb.append(", you can reduce the value by '")
sb.append(OrcConf.ROW_BATCH_SIZE.getAttribute())
sb.append("'.");
```